app/models/miq_widget_set/set_data.rb

```diff
@@ -22,15 +22,15 @@ def set_data_widgets
     private
 
     def init_set_data
-      self.set_data ||= {}
-      new_set_data ||= {}
-      self.set_data.symbolize_keys!
+      old_set_data = self.set_data&.symbolize_keys || {}
```

Comment:
Is this just avoiding the problem?  Do we want to support HWIA?  If so, I'd expect this to be `self.set_data.to_h.symbolize_keys`

Reply:
Well, it's a real database value, I have no idea if there are any readers in code depending on it being a HWIA.

Should we risk it? :)

Reply:
in my data, `MiqWidgetSet.all.map { |x| x.set_data.class }` gives

```
=> [Hash, Hash, Hash, ActiveSupport::HashWithIndifferentAccess, ActiveSupport::HashWithIndifferentAccess, Hash, Hash, ActiveSupport::HashWithIndifferentAccess, Hash, Hash, Hash, Hash, Hash, Hash, Hash, Hash, Hash, Hash, Hash, Hash]
```

So, it's probably fine to convert to Hash. Unless those Hashes are a result of the changes in #20890?

Reply:
Added, not seeing any UI code that would use string keys .. let's hope nothing breaks :)

Reply:
We probably should have a database migration to "fix" and set_datas that are storing HWIA, then.